core/src/main/java/org/infinispan/configuration/cache/EncodingConfiguration.java

```diff
@@ -18,17 +20,30 @@
  */
 public final class EncodingConfiguration implements Matchable<EncodingConfiguration>, ConfigurationInfo {
 
+   public static final AttributeDefinition<String> MEDIA_TYPE = AttributeDefinition.builder("mediaType", null, String.class).build();
+   private final AttributeSet attributes;
+
    private ContentTypeConfiguration keyDataType, valueDataType;
    private final List<ConfigurationInfo> contentTypeConfigurations;
 
    public static final ElementDefinition ELEMENT_DEFINITION = new DefaultElementDefinition(ENCODING.getLocalName());
 
-   public EncodingConfiguration(ContentTypeConfiguration keyDataType, ContentTypeConfiguration valueDataType) {
+   static AttributeSet attributeDefinitionSet() {
+      return new AttributeSet(EncodingConfiguration.class, MEDIA_TYPE);
+   }
+
+   public EncodingConfiguration(AttributeSet attributes, ContentTypeConfiguration keyDataType, ContentTypeConfiguration valueDataType) {
+      this.attributes = attributes;
       this.keyDataType = keyDataType;
       this.valueDataType = valueDataType;
       this.contentTypeConfigurations = Arrays.asList(keyDataType, valueDataType);
    }
 
+   @Override
+   public AttributeSet attributes() {
```

Comment:
I see that the toString, hashCode and equals are not based on ```attributes``` and do not include the newly added attribute.